File: app/widgets/collections/elevation_unit_combo_box.js
//= require <templates/combo_box>

JooseModule('Collections', function () {
  JooseClass('ElevationUnitComboBox', {
    isa: Templates.ComboBox,
    has: {
      method: { is: 'ro', init: 'elevation_unit' },
      collectionURI: { is: 'ro', init: function () { return Route.forPathname('length_units_path') } },
      valueMethod: { is: 'ro', init: 'id' },
      textMethod: { is: 'ro', init: 'name' },
      searchMethod:  { is: 'ro', init: 'name' },
      options: { is: 'ro', init: function () { return { parameters: { select: 'id,name' } } }}
    }
  })
});

File: app/widgets/collections/elevation_unit_combo_box.rb
class Collections::ElevationUnitComboBox < Templates::ComboBox
  def initialize options
    @collection ||= options[:collection] || fail('collection not given')
    @model_object = @collection
    @attribute_path = 'elevation_unit'
    @value_method = 'id'
    @text_method = 'label'
    @search_method = 'name'
    super
  end
end
